hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/AltoChannelConnector.java

```diff
@@ -0,0 +1,215 @@
+
+    public AltoChannelConnector(UUID clientUuid,
+                                TcpClientConnection connection,
+                                List<Integer> altoPorts,
+                                ExecutorService executor,
+                                ChannelInitializer channelInitializer,
+                                ChannelCreator channelCreator,
+                                LoggingService loggingService) {
+        this.clientUuid = clientUuid;
+        this.connection = connection;
+        this.altoPorts = altoPorts;
+        this.executor = executor;
+        this.channelInitializer = channelInitializer;
+        this.channelCreator = channelCreator;
+        this.logger = loggingService.getLogger(AltoChannelConnector.class);
+        this.altoChannels = new Channel[altoPorts.size()];
+        this.remaining = new AtomicInteger(altoPorts.size());
+    }
+
+    /**
+     * Initiates the connection attempts.
+     * <p>
+     * This call does not block.
+     */
+    public void initiate() {
+        logger.info("Initiating connection attempts to Alto channels running on ports "
+                + altoPorts + " for " + connection.toLessDescriptiveString());
+        String host = connection.getRemoteAddress().getHost();
+        int i = 0;
+        for (int port : altoPorts) {
+            int index = i++;
+            executor.submit(() -> connect(host, port, index));
+        }
+    }
+
+    private void connect(String host, int port, int index) {
+        if (connectionFailed()) {
+            // No need to try to connect if one of the channels
+            // or the connection itself is closed/failed.
+            logger.warning("The connection to Alto channel on port "
+                    + port + " for " + connection.toLessDescriptiveString()
+                    + " will not be made as either the connection or one of "
+                    + "the Alto channel connections has failed.");
+            return;
+        }
+
+        logger.info("Trying to connect to Alto channel on port " + port + " for "
+                + connection.toLessDescriptiveString());
+
+        Channel channel = null;
+        try {
+            Address address = new Address(host, port);
+            channel = channelCreator.create(address, connection, channelInitializer);
+            writeAuthenticationBytes(channel);
+            onSuccessfulChannelConnection(channel, index);
+        } catch (Exception e) {
+            logger.warning("Exception during the connection to attempt to Alto channel on port "
+                    + port + " for " + connection.toLessDescriptiveString() + ": " + e, e);
+            onFailure(channel);
+        }
+    }
+
+    private void writeAuthenticationBytes(Channel channel) {
+        // first thing we need to send is the clientUuid so this new socket can be connected
+        // to the connection on the member.
+        ClientMessage clientUuidMessage = ClientMessage.createForEncode();
```

Comment:
when will you change it similar to our generated codecs? This message does not have any correlation id, type id, etc. i think that we should just utilize existing client invocation infrastructure for this message as well.

Reply:
We will do that, I will send the PR soon and add you as the reviewer. I didn't do it here, because it was part of the another task